src/hotspot/share/opto/node.cpp

```diff
@@ -3011,7 +3011,7 @@ uint TypeNode::hash() const {
   return Node::hash() + _type->hash();
 }
 bool TypeNode::cmp(const Node& n) const {
-  return !Type::cmp(_type, ((TypeNode&)n)._type);
+  return Type::cmp(_type, ((TypeNode&)n)._type);
```

Comment:
While at it, you can replace the cast with `as_Type()`:
```suggestion
  return Type::cmp(_type, (n.as_Type()->_type);
```

Reply:
Thanks for the suggestion! I've made this change.